Automattic/pocket-casts-android
# Update horologist version (and libraries that matches this version)

## Description

This updates horologist lib to ~the latest `v0.5.14`~ the latest `0.4.x` version of `0.4.17` from the current `v0.4.10`. Below essential libraries are also upgraded to match this version:

`kotlin`: `1.7.0` to `1.9.0`
`ksp`: `1.7.10-1.0.6` to `1.9.0-1.0.13`
`compose-kotlin-compiler`: `1.3.0` to `1.5.2 `(refers [Compose Compiler to Kotlin Compatibility Map](https://developer.android.com/jetpack/androidx/releases/compose-kotlin#:~:text=1.5.2,1.9.0))
`hilt`: `2.43.2` to `2.49`
`hilt-compiler`: `1.0.0` to `1.1.0`
`media3`: `1.0.0` to `1.2.0`

In addition, we had to update the compileSdk.

#### Release notes

Horologist: https://github.com/google/horologist/releases
Kotlin: https://github.com/JetBrains/kotlin/releases?page=3
Kotlin Compiler: https://developer.android.com/jetpack/androidx/releases/compose-compiler
Hilt Android: https://mvnrepository.com/artifact/com.google.dagger/hilt-android, https://github.com/google/dagger/releases
Hilt Compiler: https://developer.android.com/jetpack/androidx/releases/hilt


## Testing Instructions
1. Smoke test the phone, automotive, and especially the watch app for any regressions since this is updating quite a few dependencies.
2. Redo the testing steps for https://github.com/Automattic/pocket-casts-android/pull/1636 to make sure that horologist is properly keeping the chips fully visible.
3. Verify that all screens which are scrollable display a scrollbar as you scroll them.

## Checklist
 
#### I have tested any UI changes...
